modular_zubbers/code/modules/true_heretic_path/code/heretic_currency/_heretic_currency.dm

```diff
@@ -26,6 +26,13 @@ GLOBAL_LIST_INIT(heretical_prefixes,generate_heretical_affixes(AFFIX_PREFIX))
 
 	return
 
+/proc/generate_heretical_quality()
+	// https://www.desmos.com/calculator/8qkyp8enui
+	var/rng = rand(1,100)
+	if(rng == 100)
+		return 6
+	return min(round(-6 + (rng * 0.056) ** 1.4,5),1)
```

Comment:
isnt exponent a fairly expensive operation in dm? i could be wrong

Reply:
> isnt exponent a fairly expensive operation in dm? i could be wrong

If you call it multiple times a second, sure, but this will only occur when someone uses currency to modify an item.